### PR TITLE
build: Add support for installing app in /app

### PR DESCRIPTION
**Pull request by hadess**
Instead of hard-coding /usr and expecting sView to be installed there,
use APP_PREFIX to allow distributors to install in other locations.

I've used this to create a Flatpak of sView.

### CODE REVIEW

File: StShared/StProcess.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ namespace {
     static const StString STCORE_NAME = StString("StCore")    + ST_DLIB_SUFFIX;
 #else
     static const StString STCORE_NAME = StString("libStCore") + ST_DLIB_SUFFIX;
-    static const StString ST_DEFAULT_PATH = "/usr/share/sView/";
+    static const StString ST_DEFAULT_PATH = "/app/share/sView/";
```
Comment (gkv311) [inline StShared/StProcess.cpp]: Seems to be APP_PREFIX was meant.

File: StShared/StFTFontRegistry.cpp [hunk 2]
```diff
@@ -112,6 +112,7 @@ StFTFontRegistry::StFTFontRegistry() {
 #else
     myFolders.add(stCString("/usr/share/fonts"));
     myFolders.add(stCString("/usr/local/share/fonts"));
+    myFolders.add(stCString(APP_PREFIX"/share/fonts"));
```
Comment (gkv311) [inline StShared/StFTFontRegistry.cpp]: Are fonts actually installed within "/app" on your system, or only sView?
There will be a dedicated patch for using fontconfig library instead of hard-coded folders which uses its own configuration file.

File: StShared/StProcess.cpp (hunk 1 above)
Comment (hadess) [reply StShared/StProcess.cpp]: Yes, mistake from an earlier version of the patch.

File: StShared/StFTFontRegistry.cpp (hunk 2 above)
Comment (hadess) [reply StShared/StFTFontRegistry.cpp]: In Flatpak `/usr/` is read-only and is what comes with the runtime. As I want to install new fonts (`DroidSansFallbackFull.ttf`, and the Nanum fonts), I need to install them with the app.

When you go to fontconfig, it'll just work as the fontconfig in Flatpak's runtime looks in `/app/share/fonts` (truncated output):
```sh
$ cat /etc/fonts/conf.d/50-flatpak.conf
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
        <dir>/app/share/fonts</dir>
        <!-- Then this so it is written when building the app -->
        <cachedir>/app/cache/fontconfig</cachedir>
</fontconfig>
```

**Comment by gkv311** [comment]
Pushed without StFTFontRegistry.cpp - fontconfig library is now used to retrieve the list of font folders on Linux. Please comment, if you find any issues.

**Comment by hadess** [comment]
Looks like it's working fine.
